pallets/handles/src/lib.rs

```diff
@@ -352,4 +281,130 @@ pub mod pallet {
 			Ok(())
 		}
 	}
+
+	impl<T: Config> Pallet<T> {
+		/// Retrieve `HandleResponse` for the specified `MessageSourceAccount`
+		/// # Arguments
+		/// * `msa_id` - The `MessageSourceAccount` to retrieve the `HandleResponse` for
+		/// # Errors
+		/// * [`Error::HandleDoesNotExist`]
+		/// # Returns
+		/// * `HandleResponse` - The `HandleResponse` for the specified `MessageSourceAccount`
+		pub fn get_handle_for_msa(msa_id: MessageSourceId) -> Option<HandleResponse> {
+			let full_handle = MSAIdToDisplayName::<T>::get(msa_id);
+			if full_handle.is_empty() {
+				return None
+			}
+
+			// convert to string
+			let full_handle_str = core::str::from_utf8(&full_handle)
+				.map_err(|_| Error::<T>::InvalidHandleEncoding)
+				.ok()?;
+			let handle_parts: Vec<&str> = full_handle_str.split(".").collect();
+			let base_handle = handle_parts[0];
+			let suffix = handle_parts[1].parse::<u16>().ok()?;
+			let canonical_handle = Self::convert_to_canonical(base_handle);
+			Some(HandleResponse {
+				base_handle: base_handle.into(),
+				suffix,
+				canonical_handle: canonical_handle.into(),
+			})
```

Comment:
Actual code thats called to get Handle given an msa (base, canonical and suffix)